src/printf.cpp

```diff
@@ -167,21 +184,66 @@ void process_printf(char*& data, const printf_descriptor_map_t& descs,
 
     // Firstly print the part of the format string up to the first '%'
     size_t next_part = format_string.find_first_of('%');
+    if (next_part > format_string.size()) {
+        data = data_end;
+    }
     printf_out << format_string.substr(0, next_part);
 
     // Decompose the remaining format string into individual strings with
     // one format specifier each, handle each one individually
     size_t arg_idx = 0;
+    std::stack<char> quotes;
+    if (format_string.size() > 0) {
+        auto curr_quote = get_quote(format_string.substr(
+            0, next_part)); // checking if curr selection is inside a " or '.
```

Comment:
Perhaps I should change this comment to "checking if curr selection has a " or '